### ISSUE TITLE
mapState inconsistent behaviour?

### ISSUE THREAD

**Issue by yokujin** [vuejs/vuex#596]
I have a module loaded in global store:
```javascript
import * as types from "../types"
export default {
	namespaced: true,
	state: {
		count: 17
	},
	mutations: {
		[types.INCREMENT] (state) {
			state.count++
		}
	},
	actions: {
		increment ({ commit }) {
			commit(types.INCREMENT)
		}
	}
}
```

And I have Vue component:
```xml
<template name="Home.vue">
	<div>
		<p id="without">{{count_non_namespaced}}</p>
		<p id="with">{{count_namespaced}}</p>
		<a href="#" @click="increment">Press me</a>
	</div>
</template>
<script charset="utf-8">
import { mapState, mapActions } from 'vuex'

export default {
    name: 'home',
    computed: {
		...mapState({
			count_non_namespaced: state => state.incrementer.count,
		}),
		...mapState('incrementer', {
			count_namespaced: state => state.count,
		}),
	},
	methods: {
		...mapActions('incrementer', [
			'increment'
		]),
	}
}
</script>
<style media="screen" lang="sass" scoped>
</style>
```

When I render this component somewhere it renders OK - both paragraphs displays `17`. When I click that `Press me` link action fires, mutation mutates state, count gets updated, template rerenders but only `p#without` shows updated value (`18`), `p#with` stays unchanged (`17`).

Did I misunderstand something? Should those computed mapState's be equivalent?

**Comment by znck** [comment]
Provide minimal reproduction code using JSFiddle/JSBin/Codepen to demonstrate your issue. You could start with this [template](http://jsfiddle.net/df4Lnuw6/) that already includes the latest version of Vue.

**Comment by ktsn** [comment]
This seems to already be fixed via #528 
@yyx990803 Can we release v2.1.2 to include the fix? This was also reported before (#565)

**Comment by znck** [comment]
Closed in favour of #565
